lib/private/Files/Conversion/ConversionManager.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OC\Files\Conversion;
+
+use OC\AppFramework\Bootstrap\Coordinator;
+use OC\SystemConfig;
+use OCP\Files\Conversion\IConversionManager;
+use OCP\Files\Conversion\IConversionProvider;
+use OCP\Files\File;
+use OCP\Files\GenericFileException;
+use OCP\Files\IRootFolder;
+use OCP\ITempManager;
+use OCP\PreConditionNotMetException;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Log\LoggerInterface;
+use RuntimeException;
+use Throwable;
+
+class ConversionManager implements IConversionManager {
+	/** @var string[] */
+	private array $preferredApps = [
+		'richdocuments',
+	];
+
+	/** @var IConversionProvider[] */
+	private array $preferredProviders = [];
+
+	/** @var IConversionProvider[] */
+	private array $providers = [];
+
+	public function __construct(
+		private Coordinator $coordinator,
+		private ContainerInterface $serverContainer,
+		private IRootFolder $rootFolder,
+		private ITempManager $tempManager,
+		private LoggerInterface $logger,
+		private SystemConfig $config,
+	) {
+	}
+
+	public function hasProviders(): bool {
+		$context = $this->coordinator->getRegistrationContext();
+		return !empty($context->getFileConversionProviders());
+	}
+
+	public function getMimeTypes(): array {
+		$mimeTypes = [];
+
+		foreach ($this->getProviders() as $provider) {
+			array_push($mimeTypes, $provider->getSupportedMimeType());
```

Comment:
```suggestion
			$mimeTypes[] = $provider->getSupportedMimeType();
```
A bit more modern (and IMO readable) syntax, but doesn't really matter in the end.